### PR TITLE
#4452 Add a helpful error message for cy.visit()

### LINKED ISSUE #4452: On cy.visit('file://...) throw err to explain they should visit relative html file
### Current behavior:

We've had a couple people try to `cy.visit('file://...)`. We support visiting `.html` files by specifying the relative path. 

- https://github.com/cypress-io/cypress/issues/4450
- https://github.com/cypress-io/cypress/pull/4293

### Desired behavior:

We can see if the user has passed in a `file://` and throw a specific error advising them to pass in the relative path to the file from their projectRoot.

### Versions

Cypress 3.3.1

### PR DESCRIPTION
**Pull request by clarmso**
- Closes #4452 

### Pre-merge Tasks

- [X] Have tests been added/updated for the changes in this PR?
- [X] Has a PR to [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation) been submitted to document any user-facing changes? https://github.com/cypress-io/cypress-documentation/issues/1989

I've added a helpful error message notifying the user that the "file:///" protocol isn't supported by Cypress, but a file can be specified by a relative path.

### CODE REVIEW

**Comment by Bkucera** [comment]
@clarmso unfortunately, if you have a `baseUrl` set, then visiting `/path/to/file` will resolve `<baseUrl>/path/to/file`, so the error message would be incorrect. I think this could get merged if we:

- [ ] add to the error message about the `baseUrl` limitation, and tell the user they would need to remove `baseUrl` if using Cypress as a static server

**Comment by clarmso** [comment]
@Bkucera Thanks for your feedback! I've added a note on baseUrl in the error message.

File: packages/driver/src/cypress/error_messages.coffee [hunk 1]
```diff
@@ -1090,6 +1090,18 @@ module.exports = {
 
         #{cmd('request')} will automatically get and set cookies and enable you to parse responses.
       """
+      
+    specify_file_by_relative_path: """
+      #{cmd('visit')} failed because the 'file://...' protocol is not supported by Cypress.
+
+      However, you can pass in the relative path to the file from the projectRoot if the baseUrl variable is unset in the configuration options.  
+      
+      If the baseUrl has been set, the baseUrl is appended to the path to the file automatically.
```
Comment (Bkucera) [inline packages/driver/src/cypress/error_messages.coffee]:
```suggestion
      If the baseUrl has been set, the baseUrl is prepended to the path to the file automatically.
```

File: packages/driver/src/cypress/error_messages.coffee [hunk 2]
```diff
@@ -1090,6 +1090,18 @@ module.exports = {
 
         #{cmd('request')} will automatically get and set cookies and enable you to parse responses.
       """
+      
+    specify_file_by_relative_path: """
+      #{cmd('visit')} failed because the 'file://...' protocol is not supported by Cypress.
+
+      However, you can pass in the relative path to the file from the projectRoot if the baseUrl variable is unset in the configuration options.
```
Comment (Bkucera) [inline packages/driver/src/cypress/error_messages.coffee]:
```suggestion
      However, you can pass in the relative path to the file from the projectRoot (Note: if the configuration value 'baseUrl' is set, the supplied path will be resolved from the 'baseUrl' instead)   
```

File: packages/driver/src/cypress/error_messages.coffee [hunk 3]
```diff
@@ -1090,6 +1090,18 @@ module.exports = {
 
         #{cmd('request')} will automatically get and set cookies and enable you to parse responses.
       """
+      
+    specify_file_by_relative_path: """
+      #{cmd('visit')} failed because the 'file://...' protocol is not supported by Cypress.
+
+      However, you can pass in the relative path to the file from the projectRoot if the baseUrl variable is unset in the configuration options.  
+      
+      If the baseUrl has been set, the baseUrl is prepended to the path to the file automatically.
```
Comment (Bkucera) [inline packages/driver/src/cypress/error_messages.coffee]:
```suggestion
```

**Comment by Bkucera** [comment]
my apologies for the ping-ponging of this review

**Comment by clarmso** [comment]
👍 Thanks for the super quick feedback!

**Pull request merged**